docusaurus/docs/how-to-guides/data-source-plugins/profile-backend-plugin.md

```diff
@@ -0,0 +1,145 @@
+
+In this section of the file, specify the `<plugin ID>`, a unique identifier, for the backend plugin you want to profile, for example, [grafana-github-datasource](https://grafana.com/grafana/plugins/grafana-github-datasource/):
+
+```ini title="custom.ini"
+[plugin.<plugin ID>]
+profiling_enabled = true
+profiling_port = 6060
+profiling_block_rate = 5
+profiling_mutex_rate = 5
+```
+
+### Confirm success
+
+Restart Grafana after applying the configuration changes. You should see a log message that indicates whether profiling was enabled. For example:
+
+```shell
+INFO [07-09|19:15:00] Profiling enabled   logger=plugin.<plugin ID> blockProfileRate=1 mutexProfileRate=1
+```
+
+### Check for debugging endpoints
+
+Check which debugging endpoints are available by browsing `http://localhost:<profiling_port>/debug/pprof`.
+
+In this file, `localhost` is used, implying that you're connected to the host where Grafana and the plugin are running. If connecting from another host, adjust as needed.
+
+## Additional endpoints
+
+There are some additional [godeltaprof](https://github.com/grafana/pyroscope-go/tree/main/godeltaprof) endpoints available for profiling. These endpoints are more suitable in a continuous profiling scenario.
+
+These endpoints are:
+
+- `/debug/pprof/delta_heap`
+- `/debug/pprof/delta_block`
+- `/debug/pprof/delta_mutex`
+
+:::note
+
+To be able to use `profiling_block_rate` and `profiling_mutex_rate`, your plugin needs to use at least [`grafana-plugin-sdk-go v0.238.0`](https://github.com/grafana/grafana-plugin-sdk-go/releases/tag/v0.238.0). Refer to [Update the Go SDK](../../create-a-plugin/develop-a-plugin/work-with-backend) for instructions on how to update the SDK.
+
+:::
+
+### The profiling_enabled endpoint
+
+Use this to enable/disable profiling. The default is `false`.
+
+### The profiling_port endpoint
+
+Optionally, customize the HTTP port where profile data is exposed. For example, use if you want to profile multiple plugins or if the default port is taken. The default is `6060`.
+
+### The profiling_block_rate endpoint
+
+Use this to control the fraction of `goroutine` blocking events that are reported in the blocking profile. The default is `0` (that is, track no events). For example, use `5` to report 20 percent of all events. Refer to https://pkg.go.dev/runtime#SetBlockProfileRate for more detailed information.
+
+:::note
+
+The higher the fraction (that is, the smaller this value) the more overhead it adds to normal operations.
+
+:::
+
+### The profiling_mutex_rate endpoint
+
+Use this to control the fraction of mutex contention events that are reported in the mutex profile. The default is `0` (that is, track no events). For example, use `5` to report 20 percent of all events. Refer to https://pkg.go.dev/runtime#SetMutexProfileFraction for more detailed information.
+
+:::note
+
+The higher the fraction (that is, the smaller this value) the more overhead it adds to normal operations.
+
+:::
```

Comment:
I'm a bit confused by these sections. Aren't those supposed to be config options instead of "endpoints" ?

Reply:
Hmm yes me too. Seems like Joe introduced this. I'll fix